src/TwigExtension.php

```diff
@@ -6,48 +7,74 @@
  * @copyright Copyright (c) 2011-2015 Josh Lockhart
  * @license   https://github.com/slimphp/Twig-View/blob/master/LICENSE.md (MIT License)
  */
+
 namespace Slim\Views;
 
+use Psr\Http\Message\UriInterface;
+use Slim\Interfaces\RouterInterface;
+
+/**
+ * Class TwigExtension
+ * 
+ * @package Slim\Views
+ */
 class TwigExtension extends \Twig_Extension
 {
     /**
-     * @var \Slim\Interfaces\RouterInterface
+     * @var RouterInterface
      */
     private $router;
 
     /**
-     * @var \Slim\Http\Uri
+     * @var UriInterface
      */
     private $uri;
 
-    public function __construct($router, $uri)
+    public function __construct(RouterInterface $router, UriInterface $uri)
     {
         $this->router = $router;
         $this->uri = $uri;
     }
 
+    /**
+     * @return string
+     */
     public function getName()
     {
         return 'slim';
     }
 
+    /**
+     * @return array
+     */
     public function getFunctions()
     {
         return [
-            new \Twig_SimpleFunction('path_for', array($this, 'pathFor')),
-            new \Twig_SimpleFunction('base_url', array($this, 'baseUrl')),
+            new \Twig_SimpleFunction('path_for', [$this, 'pathFor']),
+            new \Twig_SimpleFunction('base_url', [$this, 'baseUrl']),
         ];
     }
 
-    public function pathFor($name, $data = [], $queryParams = [], $appName = 'default')
+    /**
+     * @param $name
+     * @param array $data
+     * @param array $queryParams
+     * @param string $appName
+     * @return string
+     */
+    public function pathFor($name, array $data = [], array $queryParams = [], $appName = 'default')
     {
         return $this->router->pathFor($name, $data, $queryParams);
     }
 
-    public function baseUrl()
+    /**
+     * @param string $path
+     * @return string
+     */
+    public function baseUrl($path = '')
```

Comment:
Give the possible to use the base_url twig function with path, example:

{{ base_url('/assets/css/style.css') }}

instead

{{ base_url() }}/assets/css/style.css

Btw the old code working fine without any other modification.